packages/beacon-node/src/api/impl/beacon/blocks/index.ts

```diff
@@ -87,7 +87,10 @@ export function getBeaconBlockApi({
       const fork = config.getForkName(signedBlock.message.slot);
       let blockData: BlockInputAvailableData;
       if (isForkPostFulu(fork)) {
-        dataColumnSidecars = computeDataColumnSidecars(config, signedBlock, signedBlockOrContents);
+        const blockHash = toRootHex((signedBlock.message.body as deneb.BeaconBlockBody).executionPayload.blockHash);
+        const cachedContents = chain.producedContentsCache.get(blockHash);
```

Comment:
@g11tech please correct me if I'm misunderstanding, but I think we need to add this cache read here since `signedBlockOrContents` comes from a validator API call and won't include the cells

Reply:
yes thats correct